[develop] Remove useless mechanism to retrieve OS minor version from RHEL and Rocky

enrico-usai opened the pull request:
This mechanism was used on CentOS 7, to retrieve the OS minor version from the kernel version and use it to download the right version of the Lustre client.

This mechanism is not required for RHEL and Rocky because the minor version automatically changes when updating the kernel. This is reflected in the `/etc/os-release` file too.

This change will permit us to automatically support new minor versions when they will be available, without the need to update the internal mapping.



### Tests
* `bash kitchen.ec2.sh environment-install destroy lustre-rhel8`

I had to temporary skip installation and test of lustre client on rhel/rocky on docker.
Latest dokken/rockylinux-8 image is 8.9 and there is no Lustre client available for it.

Previously installation was passing because we were setting a fake `kernel_release` version attribute through `kitchen.docker.yml`.

### References

https://docs.aws.amazon.com/fsx/latest/LustreGuide/install-lustre-client.html

Commit:
Remove useless mechanism to retrieve OS minor version from RHEL and Rocky

This mechanism was used on CentOS 7, to retrieve the OS minor version from the kernel version
and use it to download the right version of the Lustre client.

This mechanism is not required for RHEL and Rocky because the minor version
automatically changes when updating the kernel. This is reflected in the `/etc/os-release` file too.

This change will permit us to automatically support new minor versions when they will be available,
without the need to update the internal mapping.

Signed-off-by: Enrico Usai <usai@amazon.com>

lukeseawalker commented on cookbooks/aws-parallelcluster-environment/spec/unit/resources/lustre_setup_spec.rb:
```diff
@@ -203,6 +203,7 @@ def self.setup(chef_run)
             platform: platform, version: '8',
             step_into: ['lustre']
           ) do |node|
+            node.automatic['platform_version'] = "8.#{minor_version}"
```
is the mapping for minor_version done a couple of lines above, still needed?

enrico-usai replied:
it's not strictly required for all the versions but there is a specific combination: 8.7 with a specific version of the kernel (425.13 and 425.3) that are not supported, here we're testing multiple combinations just to have more coverage.

Commit:
Clarify usage of kernel_release cluster attribute

Since the kernel version of the docker images is not in the expected pattern,
we set a fake kernel value in the `kitchen.docker.yml`to permit to install Lustre on docker.

Signed-off-by: Enrico Usai <usai@amazon.com>

Commit:
Temporary skip installation and test of lustre client on rhel/rocky on docker

Latest dokken/rockylinux-8 image is 8.9 and there is no Lustre client available for it.
We're temporary skipping installation and check, until lustre is avail.

### References

https://docs.aws.amazon.com/fsx/latest/LustreGuide/install-lustre-client.html
Signed-off-by: Enrico Usai <usai@amazon.com>

jdeamicis commented on cookbooks/aws-parallelcluster-environment/resources/lustre/lustre_centos7.rb:
```diff
@@ -104,6 +104,13 @@ def find_centos_minor_version
   os_minor_version
 end
 
+def find_kernel_patch_version
+  # kernel release is in the form 3.10.0-1127.8.2.el7.x86_64
+  kernel_patch_version = node['cluster']['kernel_release'].match(/^\d+\.\d+\.\d+-(\d+)\..*$/)
+  raise "Unable to retrieve the kernel patch version from #{node['cluster']['kernel_release']}." unless kernel_patch_version
+  kernel_patch_version[1]
+end
+
```
Can we have a unit test for this function?

enrico-usai replied:
Added

jdeamicis commented on cookbooks/aws-parallelcluster-environment/resources/lustre/partial/_install_lustre_centos_redhat.rb:
```diff
@@ -28,17 +28,12 @@
     command "yum-config-manager --setopt=\*.skip_if_unavailable=1 --save"
   end
 
+  # TODO: restore installation on docker when Lustre is available for RH8.9
```
Can't we force an 8.8 image for docker? Mmm. We are using the images from kitchen-dokken, so it looks like we can't. I wonder if we should start defining our own dokken images...

Commit:
Add unit test for find_kernel_patch_version

To load the Lustre resource and the functions defined on it,
we are using the `nothing` action.

This mechanism was already used to test for example:
- get_uuid on manage_ebs.rb
- dcv_supported? on dcv.rb

Signed-off-by: Enrico Usai <usai@amazon.com>

Commit:
Minor improvement to indentation

Signed-off-by: Enrico Usai <usai@amazon.com>
